lld/test/ELF/avr-reloc-error.s

```diff
@@ -0,0 +1,33 @@
+; REQUIRES: avr
+; RUN: llvm-mc -filetype=obj -triple=avr -mcpu=atmega328 %s -o %t.o
+; RUN: not ld.lld %t.o -o /dev/null -Ttext=0x1000 --defsym=callee=0xf00 2>&1 | \
+; RUN:     FileCheck %s --check-prefix=CHECKA --implicit-check-not=R_AVR_13_PCREL
+; RUN: not ld.lld %t.o -o /dev/null -Ttext=0x1000 --defsym=callee=0x2048 2>&1 | \
+; RUN:     FileCheck %s --check-prefix=CHECKB --implicit-check-not=R_AVR_CALL
+; RUN: not ld.lld %t.o -o /dev/null -Ttext=0x1000 --defsym=callee=0x100f 2>&1 | \
+; RUN:     FileCheck %s --check-prefix=CHECKC
+
+.section .LDI,"ax",@progbits
+
+.globl __init
+__init:
+
+; CHECKA: error: {{.*}} relocation R_AVR_7_PCREL out of range: {{.*}} is not in [-64, 63]; references 'callee'
+; CHECKB: error: {{.*}} relocation R_AVR_7_PCREL out of range: {{.*}} is not in [-64, 63]; references 'callee'
+; CHECKB: error: {{.*}} relocation R_AVR_13_PCREL out of range: {{.*}} is not in [-4096, 4095]; references 'callee'
+; CHECKC: error: {{.*}} improper alignment for relocation R_AVR_7_PCREL: {{.*}} is not aligned to 2 bytes
+; CHECKC: error: {{.*}} improper alignment for relocation R_AVR_13_PCREL: {{.*}} is not aligned to 2 bytes
+; CHECKC: error: {{.*}} improper alignment for relocation R_AVR_CALL: {{.*}} is not aligned to 2 bytes
+brne  callee
```

Comment:
brne/rjmp/jmp are independent.. It's better to use split-file and make the 3 tests completely independent and the number of lld invocations will not increase.

We can have two `rjmp` in one test case, with one out-of-range and the other barely satisfies the range requirement. This way we test the boundary of the range limit.

Reply:
Done. Thanks.